src/internal/group_api/requests.rs

```diff
@@ -194,31 +229,36 @@ pub mod group_create {
         name: Option<GroupName>,
         re_encrypted_once_value: recrypt::api::EncryptedValue,
         group_pub_key: internal::PublicKey,
-        user_id: &'a UserId,
-        member_transform_key: Option<internal::TransformKey>,
+        calling_user_id: &'a UserId,
+        user_info: Option<HashMap<UserId, (internal::PublicKey, Option<internal::TransformKey>)>>,
         needs_rotation: bool,
-    ) -> impl Future<Item = GroupBasicApiResponse, Error = IronOxideErr> + 'a {
+    ) -> impl Future<Item = GroupCreateApiResponse, Error = IronOxideErr> + 'a {
         EncryptedOnceValue::try_from(re_encrypted_once_value)
             .into_future()
             .and_then(move |enc_msg| {
+                // this currently only holds members, will need to expand for admins later
+                let req_members = user_info.map(|member| {
+                    member
+                        .into_iter()
+                        .map(|(mem_id, (pub_key, trans_key))| GroupMember {
+                            user_id: mem_id,
+                            transform_key: trans_key.unwrap().into(), // we can unwrap because we know all members had it calculated
```

Comment:
I would rather match on the `Some(trans_key)` and leave the `None` case empty (it'll be used in the next PR).

Reply:
changed to map over the option then flatten out the Nones